mlflow/server/js/src/experiment-tracking/components/ExperimentListView.tsx

```diff
@@ -25,146 +25,158 @@ import Routes from '../routes';
 import { CreateExperimentModal } from './modals/CreateExperimentModal';
 import { DeleteExperimentModal } from './modals/DeleteExperimentModal';
 import { RenameExperimentModal } from './modals/RenameExperimentModal';
+import { ExperimentPageUIState } from './experiment-page/models/ExperimentPageUIState';
 import { IconButton } from '../../common/components/IconButton';
 import { withRouterNext } from '../../common/utils/withRouterNext';
 import { ExperimentEntity } from '../types';
 
-type Props = {
+export type ExperimentListViewProps = {
   activeExperimentIds: string[];
   experiments: ExperimentEntity[];
   navigate: NavigateFunction;
+  uiState: ExperimentPageUIState;
+  setUIState: React.Dispatch<React.SetStateAction<ExperimentPageUIState>>;
 } & DesignSystemHocProps;
 
-type State = any;
+export const ExperimentListView = ({
+  activeExperimentIds,
+  experiments,
+  navigate,
+  designSystemThemeApi,
+  uiState,
+  setUIState,
+}: ExperimentListViewProps) => {
+  const list = useRef(null);
 
-export class ExperimentListView extends Component<Props, State> {
-  list: any;
-
-  state = {
-    checkedKeys: this.props.activeExperimentIds,
-    hidden: false,
+  const [state, setState] = useState({
+    checkedKeys: activeExperimentIds,
     searchInput: '',
     showCreateExperimentModal: false,
     showDeleteExperimentModal: false,
     showRenameExperimentModal: false,
     selectedExperimentId: '0',
     selectedExperimentName: '',
-  };
-
-  bindListRef = (ref: any) => {
-    this.list = ref;
-  };
+  });
 
-  componentDidUpdate = () => {
-    // Ensure the filter is applied
-    if (this.list) {
-      this.list.forceUpdateGrid();
-    }
-  };
+  // useEffect(() => {
+  //   // Ensure the filter is applied
+  //   if (list.current) {
+  //     list.forceUpdateGrid();
+  //   }
+  // }, [list]);
```

Comment:
hmm yeah in its current state, i think that this doesn't do anything. from a brief look, it seems like this was intended to make sure the `VList` below always rerenders, as it supposedly uses shallow comparison to determine whether to rerender. 

however, i think since `renderListItem` is recreated on every render, it's always going to rerender anyway. 

let's just delete this `useEffect()`.